### ISSUE TITLE
Enhancement: What about to add a converter from/to NMEA sentence?

### ISSUE THREAD

**Issue by JTrotta** [Tronald/CoordinateSharp#67]
What do you think abut that? 
All GPS devices output NMEA sentence, would be nice to add a NMEA parser and a NMEA constructor.

**Comment by Tronald** [comment]
That would be awesome. 

Are you versed on NMEA structure already? If you want to take lead on this feel free. I can do it later if not, I just have a big project at work and won't have time for a while to do the research. 

1. This should probably be contained in its own .cs file.
2. Should the Coordinate class also have a property to output NMEA sentences? What do you think?

I think it's a good idea.

**Comment by JTrotta** [comment]
I'm using this very complete parser with success: [https://www.codeproject.com/Articles/279647/NMEA-0183-sentence-parser-builder](url)
But it does not convert coordinates to NMEA. We should do conversion from Coordinate class to GPRMC  by ourselves.

**Comment by Tronald** [comment]
@JTrotta that's interesting. We could always implement a parser first, and then implement the Conversion class later. I'm ok with a version release for each.

**Comment by Tronald** [comment]
Moved to Enhancement Request List.

**Comment by Tronald** [comment]
Just an FYI to anyone interested. We reviewed this enhancement request again recently. Unfortunately, adding NMEA reliably would be quite expensive as the NMEA requires purchase of their specification. We would fall under the consumer electronics category (the highest price point).

The cost is not justifiable unfortunately as we do not expect a significant enough increase in commercial license sales to cover the expenditure. We will revisit this possibility again in a couple years and make a determination based on CoordinateSharp's current revenue generated from commercial licensing. 

We have reached out to NMEA to see if there are exceptions as we are generally free use, but don't expect any reprieve. I will continue to provide updates as I receive them.

https://www.nmea.org/nmea-0183.html
